File: src/main/java/ssg/search/collection/es/disp/ESBookMallCount.java
package ssg.search.collection.es.disp;

import io.searchbox.core.SearchResult;
import ssg.search.parameter.Parameter;
import ssg.search.result.Result;

public class ESBookMallCount extends ESMallCount {
	@Override
	public String getCollectionName(Parameter parameter) {
		return "book";
	}

	@Override
	public String getCollectionAliasName(Parameter parameter) {
		return "ESBookMallCount";
	}

	@Override
	public Result getResult(SearchResult searchResult, String name, Parameter parameter, Result result) {
		result.setBookMallCountMap(getMallCountMap(searchResult, parameter));
		return result;
	}
}

File: src/main/kotlin/ssg/front/search/api/collection/es/book/EsBookMallCount.kt
package ssg.front.search.api.collection.es.book

import org.elasticsearch.action.search.SearchResponse
import ssg.front.search.api.collection.es.common.EsMallCount
import ssg.front.search.api.dto.Parameter
import ssg.front.search.api.dto.result.Result

class EsBookMallCount : EsMallCount() {
    override var indexName = "book"

    override fun getResult(searchResponse: SearchResponse, name: String, parameter: Parameter, result: Result) {
        result.bookMallCountMap = super.getMallCountMap(searchResponse)
    }
}